src/main/java/io/pokerwars/bot/strategies/CardBasedStrategy.java

```diff
@@ -2,131 +2,135 @@
       }
 
       //with STRAIGHT or FLUSH
-      if (cardsValue == 5 || cardsValue == 6) {
-        return new Raise(
-            Math.min(minRaise * (strategyConfig.getRaiseFactor() + 2), yourChipsForRaise));
+      if (cardsStrength == 5 || cardsStrength == 6) {
+        return StraightOrFlushStrategy(strategyConfig, minRaise, yourChipsForRaise);
       }
 
       //with FULL_HOUSE, POKER or STRAIGHT_FLUSH
-      if (cardsValue >= 7) {
-        if (strategyConfig.getRaiseFactor() > 3) {
-          return new Raise(yourChipsForRaise);
-        } else {
-          return new Raise(
-              Math.min(minRaise * (strategyConfig.getRaiseFactor() + 3), yourChipsForRaise));
-        }
+      if (cardsStrength >= 7) {
+        return bestCardsStrategy(strategyConfig, minRaise, yourChipsForRaise);
       }
     }
 
     //default case
     return new Fold();
   }
 
-  private static PokerAction bluffingStrategy(GameInfo gameInfo, StrategyConfig strategyConfig) {
+  private static PokerAction bestCardsStrategy(StrategyConfig strategyConfig, Long minRaise,
+      Long yourChipsForRaise) {
+    if (strategyConfig.getRaiseFactor() > 3) {
+      return new Raise(yourChipsForRaise);
+    } else {
+      return new Raise(
+          Math.min(minRaise * (strategyConfig.getRaiseFactor() + 3), yourChipsForRaise));
+    }
+  }
+
+  private static Raise StraightOrFlushStrategy(StrategyConfig strategyConfig, Long minRaise,
+      Long yourChipsForRaise) {
+    return new Raise(
+        Math.min(minRaise * (strategyConfig.getRaiseFactor() + 2), yourChipsForRaise));
+  }
+
+  private static PokerAction threeOfAKindStrategy(StrategyConfig strategyConfig, Long minRaise,
+      Long yourChipsForRaise) {
+    if (strategyConfig.getRaiseFactor() == 1) {
+      return new Call();
+    } else {
+      return new Raise(
+          Math.min(minRaise * (strategyConfig.getRaiseFactor() + 1), yourChipsForRaise));
+    }
+  }
+
+  private static PokerAction twoPairsStrategy(StrategyConfig strategyConfig, Long minRaise,
+      Long yourChipsForRaise) {
+    if (strategyConfig.getAggressiveness() == 1) {
+      return new Fold();
+    } else if (strategyConfig.getRaiseFactor() <= 2) {
+      return new Call();
+    } else {
+      return new Raise(Math.min(minRaise * strategyConfig.getRaiseFactor(), yourChipsForRaise));
+    }
+  }
+
+  private static PokerAction highCardOrPairStrategy(StrategyConfig strategyConfig) {
+    return strategyConfig.isAggressivePlayer() ? new Call() : new Fold();
+  }
+
+  private static PokerAction canCheckOrBetStrategy(StrategyConfig strategyConfig, int cardsStrength,
+      Long minBet, Long yourChips) {
+    if (cardsStrength == 1 && strategyConfig.isNotAggressivePlayer()) {
+      return new Check();
+    } else {
+      return new Bet(Math.min(minBet * cardsStrength * strategyConfig.getRaiseFactor(), yourChips));
+    }
+  }
+
+  private static Optional<PokerAction> bluffingStrategy(GameInfo gameInfo,
```

Comment:
what's the reason to have an optional here? we should call it when we are bluffing, so there should always be a pokeraction as return, no?

Reply:
if you look at the first 3 lines of the method you can understand why... if I do it there, it means that I have to replicate that logic multiple times